packages/engine-canvas/index.js

```diff
@@ -33,14 +50,79 @@ export const run = (functionName, func, values, config) => {
     mechanic.setState(obj);
   };
 
+  let getCanvasCalled = false;
+
   func.handler({
     inputs: mechanic.values,
     mechanic: {
       frame: onFrame,
-      done: onDone,
+      done: (c) => {
+        if (!getCanvasCalled) {
+          console.warn(`Seems like you’re constructing your own canvas. @mechanic-design/engine-canvas actually provides you with a canvas that can automatically scale to the pixel density of your display.
+
+You can use it like this:
+export const handler = ({ inputs, mechanic, getCanvas }) => {
+  const { canvas, ctx } = getCanvas({ width: 1000, height: 1000 });
+  ...
+`);
+        }
+
+        onDone(c);
+      },
       state: mechanic.functionState,
       setState: onSetState,
     },
+    getCanvas: ({ width = null, height = null } = {}) => {
```

Comment:
I think it's a bit odd from the interface that we ask for the canvas and context from this function and then are supposed to give it back through `frame` and `done`. Would it make sense to keep a reference inside the engine and then use it when frame and done get called?

Reply:
Oh, I like it. I tried it and it's very nice. The only situation this is problematic is, when you also want to pass a `name` to `onDone`. So I changed `done` to accept an object as arguments. Let me know what you think.

Reply:
I do like it! It does made me realize there's no mention in the documentation about the name argument for `done`. Maybe this is an opportunity to streamline arguments for the `frame` and `done` functions and leave all of them as single argument functions on all engines. Can be done in a separate PR tho.

Reply:
I'm all in for this!